cpmpy/solvers/scip.py

```diff
@@ -0,0 +1,419 @@
+            assert isinstance(lhs, _NumVarImpl) or lhs.name == "sum" or lhs.name == "wsum", f"Unknown linear expression {lhs} on right side of indicator constraint: {cpm_expr}"
+
+            if sub_expr.name == "<=":  # XXX, docs say only this one?
+                lin_expr = self._make_numexpr(lhs)
+                if isinstance(cond, NegBoolView):
+                    self.scip_model.addConsIndicator(lin_expr <= self.solver_var(rhs),
+                                                     binvar=self.solver_var(cond._bv), activeone=False)
+                else:
+                    self.scip_model.addConsIndicator(lin_expr <= self.solver_var(rhs),
+                                                     binvar=self.solver_var(cond), activeone=True)
+
+            elif sub_expr.name == ">=": # change sign
+                if lhs.name == "sum":
+                    lhs = Operator("wsum", [[-1]*len(lhs.args), lhs.args])
+                elif lhs.name == "wsum":
+                    lhs = Operator("wsum", [[-w for w in lhs.args[0]], lhs.args[1]])
+                else:
+                    lhs = -lhs
+                self += cond.implies(lhs <= -rhs)
+
+            elif sub_expr.name == "==": # split into <= and >=
+                self += [cond.implies(lhs <= rhs), cond.implies(lhs >= rhs)]
+            else:
+                raise Exception(f"Unknown linear expression {sub_expr} name")
+
+        # True or False
+        elif isinstance(cpm_expr, BoolVal):
+            # not sure how else to do it
+            if cpm_expr.args[0] is False:
+                bv = self.solver_var(boolvar())
+                self.scip_model.addCons(bv <= -1)
+
+        # a direct constraint, pass to solver
+        elif isinstance(cpm_expr, DirectConstraint):
+            cpm_expr.callSolver(self, self.scip_model)
+
+        else:
+            raise NotImplementedError(cpm_expr)  # if you reach this... please report on github
+
+      return self
+
+    def solveAll(self, display=None, time_limit=None, solution_limit=None, call_from_model=False, **kwargs):
+        """
+            Compute all solutions and optionally display the solutions.
+
+            This is the generic implementation, solvers can overwrite this with
+            a more efficient native implementation
+
+            Arguments:
+                - display: either a list of CPMpy expressions, OR a callback function, called with the variables after value-mapping
+                        default/None: nothing displayed
+                - time_limit: stop after this many seconds (default: None)
+                - solution_limit: stop after this many solutions (default: None)
+                - call_from_model: whether the method is called from a CPMpy Model instance or not
+                - any other keyword argument
+
+            Returns: number of solutions found
+        """
+
+        warnings.warn("Solution enumeration is not implemented in PyScipOPT, defaulting to CPMpy's naive implementation")
```

Comment:
So I mentioned this is an email. I think it's just that very few MIP practitioners actually need such functionality. To mimic most of this:
- Call optimize for the ```self.scip_model```.
- After solving is completed. Call ```self.scip_model.getSols()```. 
The potential problems with this is that SCIP automatically deletes the "worst" solutions in storage after it has found x many. This is by default 10. See ```limits/maxsol``` and ```limits/maxorigsol```. So to save memory SCIP will at times remove older solutions that it knows are suboptimal because it has clearly better ones. The number of solutions found in total can be retrieved with ```scip_model.getNSolsFound()```, but unless you increase the limits you will not be able to retrieve all of them.